datafusion/physical-plan/src/insert.rs

```diff
@@ -73,6 +73,8 @@ pub struct FileSinkExec {
     sink_schema: SchemaRef,
     /// Schema describing the structure of the output data.
     count_schema: SchemaRef,
+    /// Optional required sort order for output data.
+    sort_order: Option<Vec<Option<Vec<PhysicalSortRequirement>>>>,
```

Comment:
Since FileSink can have only a single input, I think it only needs a single sort order per [`required_input_order`](https://docs.rs/datafusion/latest/datafusion/physical_plan/trait.ExecutionPlan.html#method.required_input_ordering)

In other words, I think this could be simplified to 

```suggestion
    sort_order: Option<Vec<PhysicalSortRequirement>>>,
```

And then adjust required_input_order appropriately

Reply:
This makes sense. Will do!

Reply:
Just pushed an update with this change